src/scoring/segmentation.py

```diff
@@ -0,0 +1,153 @@
+            # Crop the image and its segmentation based on
+            # the previously found road centers
+            images, pickles = self._crop_panoramic_images(image, segmentation)
+            # Calculate the Green View Index (GVI) for the cropped segmentations
+            GVI, segment_scores = self._get_GVI(pickles)
+            return GVI
+        except Exception as ex:
+            log.error(ex)
+
+    def _segment_images(self, image: Image):
+        """
+        Applies semantic segmentation to an image
+        Args:
+            image: to apply segmentation to
+
+        Returns: the segmentation as a torch tensor
+
+        """
+        # Preprocess the image using the image self.processor
+        inputs = self.processor(images=image, return_tensors="pt")
+        # Perform a forward pass through the self.model to obtain the segmentation
+        with torch.no_grad():
+            # Check if a GPU is available
+            if torch.cuda.is_available():
+                # Move the inputs to the GPU
+                inputs = {k: v.to("cuda") for k, v in inputs.items()}
+                # Perform the forward pass through the self.model
+                outputs = self.model(**inputs)
+                # Post-process the semantic segmentation outputs
+                # using the processor and move the results to CPU
+                segmentation = self.processor.post_process_semantic_segmentation(
+                    outputs, target_sizes=[image.size[::-1]]
+                )[0].to("cpu")
+            else:
+                # Perform the forward pass through the self.model
+                outputs = self.model(**inputs)
+                # Post-process the semantic segmentation outputs using the processor
+                segmentation = self.processor.post_process_semantic_segmentation(
+                    outputs, target_sizes=[image.size[::-1]]
+                )[0]
+        return segmentation
+
+    def _crop_panoramic_images(self, image: Image, segmentation: torch.Tensor):
+        """
+        Crops the panoramic image and its segmentation based on road centers
+        Args:
+            image: opened image
+            segmentation: segmentation of the image
+
+        Returns: list of cropped images and their segmentations
+
+        """
+        width, height = image.size
+        w4 = int(width / 4)
+        h4 = int(height / 4)
+        hFor43 = int(w4 * 3 / 4)
+        images = []
+        pickles = []
+        # Crop the panoramic image based on road centers
+        for w in range(4):
+            x_begin = w * w4
+            x_end = (w + 1) * w4
+            cropped_image = image.crop((x_begin, h4, x_end, h4 + hFor43))
+            cropped_segmentation = segmentation[h4 : h4 + hFor43, x_begin:x_end]
+            images.append(cropped_image)
+            pickles.append(cropped_segmentation)
+        return images, pickles
+
+    def _get_GVI(self, segmentations: list[torch.Tensor]):
```

Comment:
```suggestion
    def _get_gvi(self, segmentations: list[torch.Tensor]):
```

Code style preference.

Reply:
Will do, but also, code style issues should be caught by linter or autoformatter rules, not by manual review. For the future, let's review the Ruff rules and see which extra ones we want to add, such as variable capitalization.